server/views/media_picker.py

```diff
@@ -148,65 +154,74 @@ def _cached_featured_collection_list(tag_id_list):
             query_media_ids = " ".join([str(m) for m in media_ids])
             query_media_ids = re.sub(r'\[*\]*', '', str(query_media_ids))
             query_media_ids = " media_id:({})".format(query_media_ids)
             query += '('+query_media_ids+')'
 
         # conjunction
-        if len(media_ids) > 0 and (len(tags_ids) > 0 or len(custom_ids) > 0):
+        if len(media_ids) > 0 and (len(tags_ids) > 0 or len(custom_collection) > 0):
             query += " OR "
 
         # add in the collections they specified
-        if len(tags_ids) > 0:
+        if len(tags_ids) > 0: # this format is a list of tags_id_medias
             tags_ids = tags_ids.split(',') if isinstance(tags_ids, str) else tags_ids
             query_tags_ids = " ".join([str(t) for t in tags_ids])
             query_tags_ids = re.sub(r'\[*\]*', '', str(query_tags_ids))
             query_tags_ids = " tags_id_media:({})".format(query_tags_ids)
-            if len(custom_ids) == 0:
+            if not len(custom_collection) > 0:
                 query += '('+query_tags_ids+')'
             else:
                 query += '(' + query_tags_ids
 
-        # grab any custom collections and turn it into a boolean tags_id_media phrase
-        if len(custom_ids) > 0:
-            query_custom_ids = custom_collection_as_solr_query(custom_ids)
-            if len(tags_ids) > 0:
-                query = "{} OR {} )".format(query, query_custom_ids)  # OR all the sets with the other Collection ids
-            else:
-                query += query_custom_ids  # add the sets to the query (the OR was added before)
+        # grab any custom collections (there may be multiple) and turn it into a boolean tags_id_media phrase
+        if len(custom_collection) > 0: #this format is for each metadata tag, each sub-list is OR'd together, and every list is AND'ed together
+            # we may not have tags_id_media but we may have a keyword
+            #how many?
+            custom_collection_dict = json.loads(custom_collection)
+            has_custom_collections, custom_collection_string  = custom_collection_as_solr_query(custom_collection_dict)
+            query = "{} {}".format(query, custom_collection_string)
+
+            # else nothing
+        # add the sets to the query (the OR was added before)
+            query += ')'
         query += ')'
-
     return query
 
 
-def custom_collection_as_solr_query(custom_ids_str):
-    if (custom_ids_str is None) or (len(custom_ids_str) == 0):
-        return ''
-    custom_ids_dict = json.loads(custom_ids_str)
-    query_custom_ids = ''
-    for sets_of_tags in custom_ids_dict:  # for each custom collections
-        custom_tag_groups = json.loads(sets_of_tags['tags_id_media'])  # expect tags in format [[x, ...], ...]
-        custom_sets = []
-        for tag_grp in custom_tag_groups:
+def custom_collection_as_solr_query(custom_coll_dict):
+    if (len(custom_coll_dict) == 0):
+        return False
```

Comment:
This method appears to either return a tuple or just a single boolean. Can be be consistent?

Also, it looks like the boolean isn't needed--doesn't seem to be used anywhere. If so, can it be removed?

Reply:
it is implicitly used. I updated the code to make it explicitly use the boolean, and to return both boolean and query string